canvas_sdk/v1/data/base.py

```diff
@@ -91,3 +123,52 @@ def codings(value_set: Type["ValueSet"]) -> tuple[tuple[str, set[str]]]:
                 if i[0] in values_dict
             ),
         )
+
+
+class TimeframeLookupQuerySetProtocol(QuerySetProtocol):
+    """A typing protocol for use in TimeframeLookupQuerySetMixin."""
+
+    @property
+    @abstractmethod
+    def timeframe_filter_field(self) -> str:
+        """A protocol method for timeframe_filter_field."""
+
+    ...
+
+
+class TimeframeLookupQuerySetMixin(TimeframeLookupQuerySetProtocol):
+    """A class that adds queryset functionality to filter using timeframes."""
+
+    @property
+    def timeframe_filter_field(self) -> str:
+        """Returns the field that should be filtered on. Can be overridden for different models."""
+        return "note__datetime_of_service"
+
+    def within(self, timeframe: "Timeframe") -> models.QuerySet:
```

Comment:
I agree with the more specific naming, but there's also carrying over methods developers are used to using from the workflow kit.

Reply:
I suppose I would say that given how dissimilar plugins are to the workflow kit, keeping names isn't something I would value very highly. Converting a protocol to a plugin is a lot of work, and it doesn't really have anything to do with names. We're looking for functional parity, not compatibility per se.

Since this is a design issue that will impact users, it's probably worth soliciting more opinions on this.

@aduane @mbiannaccone Please chime in if you have thoughts about name clashes between mixins that we are using to create composable ValueSets.

Reply:
I can see your position, @csande , but let's see if it becomes a problem, and if it does, we can do some change management to solve it.